src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -396,19 +397,20 @@ public void execute(BuildScanExtension buildScan) {
             }
 
             if (isNotEmpty(gitRepo) && isNotEmpty(gitCommitId)) {
-                if (gitRepo.contains("github.com/") || gitRepo.contains("github.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)github\\.com[/|:](.*)").matcher(gitRepo);
-                    if (matcher.matches()) {
-                        String rawRepoPath = matcher.group(2);
-                        String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
-                        buildScan.link("Github source", "https://github.com/" + repoPath + "/tree/" + gitCommitId);
-                    }
-                } else if (gitRepo.contains("gitlab.com/") || gitRepo.contains("gitlab.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)gitlab\\.com[/|:](.*)").matcher(gitRepo);
+                Optional<String> gitHubUrl = envVariable("GITHUB_SERVER_URL", providers);
+                Optional<String> gitHubRepository = envVariable("GITHUB_REPOSITORY", providers);
+                if (gitHubUrl.isPresent() && gitHubRepository.isPresent()) {
+                    buildScan.link("GitHub source", gitHubUrl + "/" + gitHubRepository + "/tree/" + gitCommitId);
```

Comment:
Did you really test this? It seems to me that you have an implicit conversion from an Optional to a String here, which would not lead to a properly rendered URL.

Reply:
Yeah that was definitely an oversight - I'll get that fixed.

Reply:
Fixed. [Here](https://ge.solutions-team.gradle.com/s/c62popbv5t6d6) is a build scan from a build run in GitHub Actions (verified by the presence of the `CI` tag and `GitHub Actions build` link) that generates a correct `GitHub source` link.

I published the fix as CCUD version `1.10.2` to the staging portal, and we can verify [here](https://ge.solutions-team.gradle.com/s/c62popbv5t6d6/plugins#com.gradle.common-custom-user-data-gradle-plugin-20) that version `1.10.2` was used by that build scan.